core/EAP7-1266_WFCORE-4314_enchance_keystore_commands_alias.adoc

```diff
@@ -0,0 +1,343 @@
+
+== Overview
+
+ Elytron keystore expose `read` commands to access content:
+ * read-alias
+ * read-aliases
+
+ There are two problems with those commands. The `read-alias` command is too verbose by default. When alias is displayed after read command, output include hex format of public-key and encoded fields. This is usually not something required if one inspects key and it clouds output with dump of unreadable string of hex digits.
+ Second problem is that read-aliases is in fact "read-aliases-names". In order to read content of keystore one has to read alias names with `read-aliases` and issue `read-alias` for each name. To make it more user friendly, simple changes are required:
+ * VERBOSE - new parameter for both commands. Default value would be `false`. This parameter control inclusion of bulk hex fields of key(public-key and encoded)
+ * RECURSIVE - new parameter for `read-aliases`. Default value `false`. If set otherwise, `read-aliases` will read recurse into keys, displaying more than names.
+
+CAUTION: default value of VERBOSE parameter, in current implementation is `true`. Explicity definition of parameter with default value of `false` is a change in behavior.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4314
+
+=== Related Issues
+
+=== Dev Contacts
+
+* mailto:mailto:{email}[{author}]
+* mailto:darran.lofthouse@redhat.com[Darran Lofthouse]
+* mailto:fjuma@redhat.com[Farah Juma]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+ * wildfly-core
+ * HAL(possibly)
+
+=== Other Interested Projects
+
+ n/a
+
+== Requirements
+=== Hard Requirements
+
+ Make read commands of elytron keystore tad more user friendly. Include changes in HAL if it supports original syntax.
+
+=== Nice-to-Have Requirements
+
+
+
+=== Non-Requirements
+
+=== Design
+
+Preparation:
+[source]
+--
+/subsystem=elytron/key-store=serverKS:add(path=server.keystore.jsk,relative-to=jboss.server.config.dir,credential-reference={clear-text=secret}, type=JKS)
+cd subsystem=elytron/key-store=serverKS
+:generate-key-pair(alias=server1,distinguished-name="CN=www.example.com")
+[standalone@localhost:9990 key-store=serverKS] :generate-key-pair(alias=server2,distinguished-name="CN=www.example.com")
+[standalone@localhost:9990 key-store=serverKS] :generate-key-pair(alias=backup,distinguished-name="CN=www.example.com")
+--
+
+
+`read-aliases` - no change in default output.
+[source]
+--
+[standalone@localhost:9990 key-store=serverKS] :read-aliases
+{
+    "outcome" => "success",
+    "result" => [
+        "backup",
+        "server2",
+        "server1"
+    ]
+}
+--
+
+
+`read-aliases` - inclusion of single parameter does not change output.
+[source]
+--
+[standalone@localhost:9990 key-store=serverKS] :read-aliases(verbose=true)
```

Comment:
Should this be `verbose=false`?

Reply:
This is just to show that it, alone, should not trigger change in output.